hilti/toolchain/src/ast/scope.cc

```diff
@@ -11,20 +11,12 @@
 
 using namespace hilti;
 
-void Scope::insert(const ID& id, NodeRef&& n) {
-    assert(n && n->isA<Declaration>());
-    const auto& d = n->as<Declaration>();
-    auto& nodes = _items[std::string(id)];
-
-    // Filter out duplicates
-    for ( const auto& i : nodes ) {
-        if ( i && i->as<Declaration>() == d )
-            return;
-    }
-
-    nodes.push_back(std::move(n));
+bool Scope::ItemCmp::operator()(const NodeRef& a, const NodeRef& b) const {
+    return a->as<Declaration>().meta().location() < b->as<Declaration>().meta().location();
```

Comment:
I'm not sure this is safe: we don't guarantee that nodes must have a location, and different nodes could also end up with the same location (locations are not only coming in from source code, but also programmatically).